sigmf/sigmffile.py

```diff
@@ -630,9 +630,11 @@ def get_sigmf_filenames(filename):
     Returned as dict with 'data_fn', 'meta_fn', and 'archive_fn' as keys.
 
     Keyword arguments:
-    filename -- the SigMF filename
+    filename -- the SigMF dataset filename (with or without .sigmf* extension)
```

Comment:
Its not clear to me that the filename here _should_ be supported without the .sigmf* extension. The uses in `tofile` and `fromfile` are a bit ambiguous so a resolution should clarify the permitted filenames to those functions also. I read the intention here as "needs to be a valid sigmf filename".

Reply:
In my mind, the `SigMFFile` object is a way of interacting with the entire dataset (metadata _and_ samples), so semantically I think it makes more sense for the user to refer to a (non-archive) dataset by its base name rather than either the `.sigmf-meta` filename or `.sigmf-data` filename specifically.

Reply:
Worth noting, here, is that once #190 gets merged, the Dataset file can be specified from within the Metadata file, and it may not share the same basename. 

From a **code** perspective, though, I believe @ke8ctn's understanding of how it's treated is correct, here. Really need @Teque5 to weigh-in, though, as the original author.

Reply:
Oh, good point. With #190 I think it will be cleaner to require a `.sigmf-meta` or `.sigmf` filename, since we'' still be able to infer the dataset filename from the metadata (or its filename), but not the other way around. That has me leaning more toward requiring the full metadata filename with its extension.

Also, I just realized that all this time I've been saying "SigMF **dataset**" when I should have been saying "SigMF **recording**".